pkgs/development/libraries/wolfssl/default.nix

```
@@ -104,4 +104,9 @@ in
       "tls13"
     ];
   };
+
+  wolfssl-haproxy = buildWolfSSL {
+    variant = "haproxy";
+    extraOptions = [ "quic" ];
+  };
```

```suggestion
  wolfssl-haproxy = wolfssl.overrideAttrs (oldAttrs: {
    pname = "wolfssl-haproxy";

    configureFlags = oldAttrs.configureFlags ++ [
      "--enable-haproxy"
      "--enable-quic"
    ];
  });
```
with this simple change we can almost completely discard the previous commit.